unity-renderer/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -2320,6 +2320,10 @@ protected virtual IEnumerator ConstructMaterial(GLTFMaterial def, int materialIn
             for (int i = 0; i < 2; i++)
             {
                 string materialCRC = material[i].ComputeCRC().ToString();
+                if (def.Name.Contains("Hair"))
+                    materialCRC += "Hair";
+                else if (def.Name.Contains("Skin"))
+                    materialCRC += "Skin";
```

Comment:
Maybe we can disable the material cache altogether for avatars instead of doing this. It should be more clean.

In fact I forgot that I modified this line:
https://github.com/decentraland/unity-renderer/blob/203e0dfb8867af122688dae11dbee18f4988b061/unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/WearableController.cs#L67
Formerly it was set as `CACHE_SHADERS`.

Reply:
I modified the line because I thought that we copy the material before changing its color. Now I look at the code and it seems that i was mistaken. As `MapSharedMaterialsRecursively` is used directly on the mats.

Reply:
I didnt even try that because I didnt want to degrade the performance. I can profile both solutions to find degradation in performance.

Reply:
I don't have enough context but I agree with Brian. At least we should add a comment explaining this functionality. Using the gltf scene importer for something so specific to the avatars is an overkill imo

Reply:
I think that replacing this fix for `CACHE_SHADERS` in `WearableController` shouldn't degrade performance significantly. With both approaches the outcome is to create a new Material. The only difference is that we will clone a few more materials, but I think is worth it as the solution would be more maintainable, and the gltf importer shouldn't know about the specifics of the world.

Reply:
Okay! gonna do it!